best-practices.md

```diff
@@ -155,9 +179,24 @@ Self-contained catalogs are not just for offline use, however - they are designe
 on the cloud in object storage. They just aim to ease the burden of publishing, by not requiring lots of updating of links. 
 Adding a single `self` link at the root is recommended for online catalogs, turning it into a 'relative published catalog', as detailed below. This anchors it in an online location and enable provenance tracking.
 
-### Versioning for Catalogs
+### Published Catalogs
```

Comment:
So what's the recommended best practice here?

Reply:
What do you mean? You want it to say like 'if you are publishing a catalog online it should be one of the following'? Or something like that?

Reply:
This chapter reads like a specification, not a best practice. It sounds like "you must use either absolute or relative catalogs" although there could be catalogs that mix relative and absolute. I think it should sound more like "it is recommended to not mix absolute and relative and stick to one of the two following types ... due to the reasons ..." or so...

Reply:
Ok, just updated this, let me know if it hits the mark.

Reply:
Yes, sounds good. Although not I'm realizing that links are always mixed in the "relative" case. Once you add an external link to whatever resource, it's always mixed. It says "relative links for everything", but then you can't put any external links? That seems confusing...

Reply:
Hrmm... I'm going back and forth on this. I see the relative published catalog as one that is from a self-contained catalog, and a self-contained catalog I believe should only have relative links, so it's actually fully portable. Like that manual should be included locally, not be external. 

And then a relative published one is one of those that changes its root. So I think I would say if it doesn't have relative links for everything then it is no longer a 'relative published catalog'. I don't think it's that you *can't* put any external links. But once you do you're no longer a 'relative published catalog'. Which I think is fine, but it is slightly more confusing to clients. 

I'd say the one thing that could make sense to me given that is to recommend a bit less strongly here, since it is likely that will be catalogs that are a mix. But I think the best practice for a published catalog is one of the two articulated.

Reply:
So no way to link to the homepage of the provider, DOIs, licenses, for example? That seems of limited use?!
I guess we could relax it that the rel types root, parent, child, item, collection (i.e. all that point to documents specified by STAC) must be relative, but others (i.e. license, doi, about, alternate) can be absolute.

Reply:
I do think that many people using self-contained catalogs aren't really linking out much, like not bothering with provider or license, let alone DOI's. But I think it's a fair point that this is about the structural links, will clarify.

Reply:
Ok, took a crack at this. Will leave this conversation open, as I'd like to merge, and can do another PR if more tweaks are needed. I realized that the 'self-contained catalog' already did a better job with this, so I tweaked it to be a bit more explicit about the rel types allowed, etc. And then just had the relative published catalog refer to it.